### PR TITLE
Basic testable comments, they don't hit the server, just local, for a…

### PR DESCRIPTION
**Pull request by jaredpetker**
…dmins

### CODE REVIEW

**Comment by jaredpetker** [comment]
@cvializ initial go at comments if you wanna check them out, it's all local and just for admins for testing

**Comment by camhashemi** [comment]
looks great!

File: kifi-backend/modules/shoebox/angular/src/common/directives/comments/comment/comment.tpl.html [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<div class="kf-keep-comment-container">
+
+  <img src="{{comment.user|pic:100}}" class="kf-keep-comment-img"/>
+  <div class="kf-keep-comment-content">
+    <div>
+        <span class="kf-keep-comment-commenter-text">
+      <a class="kf-keep-comment-commenter-name">{{::comment.user|name}}</a>{{comment.text}}</span>
```
Comment (cvializ) [inline kifi-backend/modules/shoebox/angular/src/common/directives/comments/comment/comment.tpl.html:7]: 1. We need to use `ng-bind` so users don't see {{::comment.user|name}} if the page renders slowly.
2. `<a>` tags need an `href` or `ui-sref`. This should be a `<button class="kf-button-textonly">` if it won't ever link to anything

File: kifi-backend/modules/shoebox/angular/src/common/directives/comments/comments.tpl.html [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<div class="kf-keep-comments-container">
+  <div class="kf-keep-comments">
+    <div kf-keep-comment comment="comment" ng-repeat="comment in comments"
+         ng-class="{'kf-keep-comments-comment-v-spacing': true}"></div>
+
+    <div class="kf-keep-comments-input-container" >
+      <img src="{{me|pic:100}}" class="kf-keep-comments-input-user-img"/>
```
Comment (cvializ) [inline kifi-backend/modules/shoebox/angular/src/common/directives/comments/comments.tpl.html:7]: `ng-src`

File: kifi-backend/modules/shoebox/angular/src/common/directives/comments/comments.styl [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  width: 100%
+  height: auto
+  flex-direction: row
+  justify-content: space-around
+
+  .kf-keep-comments
+    display: flex
+    flex-direction: column
+    width: 90%
+    border-left: solid 1px #3e495533
```
Comment (cvializ) [inline kifi-backend/modules/shoebox/angular/src/common/directives/comments/comments.styl:12]: protip, if you want to repeat less you can use the shorthand `border: solid 1px #color` then remove the side you don't need with `border-top: 0`

File: kifi-backend/modules/shoebox/angular/src/common/directives/comments/comments.tpl.html [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+      <div class="kf-keep-comments-input ng-pristine ng-valid ng-touched" ng-keydown="keydown($event)" placeholder="write a comment..."
```
Comment (cvializ) [inline kifi-backend/modules/shoebox/angular/src/common/directives/comments/comments.tpl.html:8]: why did you add `ng-pristine` `ng-valid` and `ng-touched`? angular adds these, it's usually not something a developer uses

File: kifi-backend/modules/shoebox/angular/src/common/directives/comments/comment/comment.tpl.html [hunk 5]
```diff
@@ -0,0 +1,14 @@
+<div class="kf-keep-comment-container">
+
+  <img src="{{comment.user|pic:100}}" class="kf-keep-comment-img"/>
```
Comment (cvializ) [inline kifi-backend/modules/shoebox/angular/src/common/directives/comments/comment/comment.tpl.html:3]: ng-src, or the browser will make a request to the server for kifi.com/{{comment.user|pic:100}}

File: kifi-backend/modules/shoebox/angular/src/common/directives/comments/comments.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+  .directive('kfKeepComments', [ 'profileService',
+    function (profileService) {
+      return {
+        restrict: 'A',
+        scope: {
+          keep: '='
+        },
+        templateUrl: 'common/directives/comments/comments.tpl.html',
+        link: function (scope, element) {
+          var inputDiv = element[0].querySelector('.kf-keep-comments-input');
```
Comment (cvializ) [inline kifi-backend/modules/shoebox/angular/src/common/directives/comments/comments.js:14]: Why did you choose `contenteditable` over a simpler `<textarea class="kf-textbox>`? `contenteditable`s can be difficult to wrangle when bugs appear

**Comment by cvializ** [comment]
it looks really slick :sunglasses:

File: kifi-backend/modules/shoebox/angular/src/common/directives/comments/comment/comment.tpl.html (hunk 1 above)
Comment (jaredpetker) [inline kifi-backend/modules/shoebox/angular/src/common/directives/comments/comment/comment.tpl.html:7]: it will have an href, I just haven't done it yet.

File: kifi-backend/modules/shoebox/angular/src/common/directives/comments/comments.tpl.html (hunk 4 above)
Comment (jaredpetker) [inline kifi-backend/modules/shoebox/angular/src/common/directives/comments/comments.tpl.html:8]: they were on a previous one I found for checking how some inputs were done, they seemed fishy when copied them. will remove.

File: kifi-backend/modules/shoebox/angular/src/common/directives/comments/comments.js (hunk 6 above)
Comment (jaredpetker) [inline kifi-backend/modules/shoebox/angular/src/common/directives/comments/comments.js:14]: well, the textarea had the stupid grabber resizer, and wasn't sure how to get textarea to resize automatically. I'm not opposed to using whichever, but when I looked around it seemed like people were using divs potentially for comment box type things. If you can point me to doing the textareas properly then can move to that easily.

Comment (cvializ) [inline kifi-backend/modules/shoebox/angular/src/common/directives/comments/comments.js:14]: ah I see facebook is using the contenteditable. We can try it here and see how it works out; I hate having to make things resize themselves too

**Comment by cvializ** [comment]
You'll definitely want to fix the src -> ng-src. my console is filling up with 500s
screen shot 2015-11-24 at 1 22 51 pm

**Comment by jaredpetker** [comment]
Yea, I fixed them all,should be fine after I release.

File: kifi-backend/modules/shoebox/angular/src/common/directives/comments/comments.js (hunk 6 above)
Comment (jaredpetker) [inline kifi-backend/modules/shoebox/angular/src/common/directives/comments/comments.js:14]: Yea facebook's was fun to look at, but yea I think I waded through some of the nuances already (or at least some) so we can see how it goes.
